module/VuFind/src/VuFind/Log/Logger.php

```diff
@@ -51,110 +52,189 @@
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Site
  */
-class Logger extends BaseLogger
+
+
+class Logger implements PsrLoggerInterface
 {
     /**
      * Is debug logging enabled?
      *
-     * @var bool
+     * @var bool`
      */
-    protected $debugNeeded = false;
+    protected bool $debugNeeded = false;
 
     /**
      * User IP address reader
      *
      * @var UserIpReader
      */
-    protected $userIpReader;
+    protected UserIpReader $userIpReader;
+    private MonologLogger $monolog;
+
+    private const LEVEL_MAP = [
```

Comment:
We typically make everything protected rather than private, to support extension/overriding of services. Unless there's a strong reason to make these private, I'd recommend protected instead:

```suggestion
    protected MonologLogger $monolog;

    protected const LEVEL_MAP = [
```